CMakeLists.txt

```diff
@@ -32,6 +32,30 @@ if (NOT CMAKE_CONFIGURATION_TYPES)
     set(CMAKE_CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo;MinSizeRel" CACHE STRING "Configs")
 endif()
 
+# Determine UNIX distribution
+if(NOT WIN32)
+    file(READ "/etc/os-release" ETC_ISSUE)
+    string(REGEX MATCH "Debian|Ubuntu|CentOS|RHEL|SLES" DIST ${ETC_ISSUE})
+    set(USE_TBB Off)
+
+    if (DIST STREQUAL "Debian")
+        message(STATUS "Debian OS detected!")
+        add_compile_definitions(DEBIAN_DISTRO)
+        set(USE_TBB On)
+    elseif(DIST STREQUAL "Ubuntu")
+        message(STATUS "Ubuntu OS detected!")
+        add_compile_definitions(DEBIAN_DISTRO)
+        set(USE_TBB On)
+    elseif(DIST STREQUAL "CentOS")
+        message(STATUS "CENTOS detected!")
+    elseif(DIST STREQUAL "Red Hat")
+        message(STATUS "Red Hat detected!")
+    elseif(DIST STREQUAL "SUSE")
+        message(STATUS "SLES detected!")
+    endif()
```

Comment:
Is it useful to flag the` else `condition? Say, for debugging purposes?

Reply:
Didn't think to add it as I wanted to be explicit where this is run. This change arose as originally we were having issues with TBB in other distros outside of ubuntu and this is why this change set was shelved for the time being. I know Paul got SLES working with TBB so its why I've resurrected this PR

Reply:
Appreciate your resurrecting the PR. Please add an else clause just to indicate it as an error condition, if that is logically correct. Thanks.